tools/project_manager/editproject.php

```diff
@@ -599,6 +603,40 @@ function set_from_post()
             }
         }
 
+        $this->custom_chars = @$_POST['custom_chars'];
+        if($this->custom_chars)
+        {
+            $codepoints = utf8_codepoints_combining($this->custom_chars);
+
+            // all characters must be unique
+            if($codepoints != array_unique($codepoints))
+            {
+                $errors .= _("The set of custom characters must be unique.")."<br>";
+            }
+
+            // all characters must be normalized
+            if(count(get_nonnormalized_codepoints($codepoints)))
+            {
+                $errors .= _("Custom characters must be in NFC normalized form.")."<br>";
+            }
+
+            // only allow 32 characters
+            if(count($codepoints) > 32)
+            {
+                $errors .= _("A maximum of 32 custom characters are allowed.")."<br>";
+            }
+
+            // prevent disallowed characters from being added
+            $disallowed_codepoints = array_intersect(get_disallowed_codepoints(), $codepoints);
+            if($disallowed_codepoints != [])
```

Comment:
Maybe add the space to the disallowed codepoints? This is what the picker tile looks like for a space: 
<img width="41" alt="image" src="https://user-images.githubusercontent.com/36800239/87844717-9a1cb680-c885-11ea-974d-58dee7974cf8.png">

Reply:
We could special-case a space, but we can't add it to `get_disallowed_codepoints()` because it's allowed.

Reply:
Oops. Right. Silly me. I was focused on "disallowed for picker sets" not the fact that it would be disallowing the space completely. Yeah, bad idea.